### PR TITLE
Fix training IR bug by changing passes order

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #138517
* #138516
* #138261
* __->__ #138292
* #138266

Inserting runtime_assertions cause gm to have different names but the graph signature was populated earlier. To avoid this kind of errors in the future, I refactored these steps into a helper function. 

Differential Revision: [D64576251](https://our.internmc.facebook.com/intern/diff/D64576251)

### CODE REVIEW

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D64576251).

File: torch/_export/utils.py [hunk 1]
```diff
@@ -529,6 +533,34 @@ def nodes_filter(nodes: List[torch.fx.Node], node_call_back) -> List[torch.fx.No
...
+                insert_deferred_runtime_asserts(
+                    gm,
+                    shape_env,
+                    f"exported program: {first_call_function_nn_module_stack(gm.graph)}",
+                    export=True,
+                )
+    # update output specs
+    gm.recompile()
+    graph_signature.user_outputs = _graph_output_names(gm)
+    return gm, graph_signature
```
Comment (avikchaudhuri) [inline torch/_export/utils.py:561]: why return gm, graph_signature if you're not using it below?

Comment (tugsbayasgalan) [reply torch/_export/utils.py:561]: Yeah it can return None. But maybe i should return and replace to make it more clear

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D64576251).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D64576251).

**Review by avikchaudhuri — APPROVED** [review]

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D64576251).

**Pull request closed without merging**